apps/src/blockly/addons/cdoConnectionChecker.js

```diff
@@ -0,0 +1,58 @@
+import GoogleBlockly from 'blockly/core';
+import {customConnectionBlockTypes} from './cdoConstants';
+
+/**
+ * A connection checker that imposes stricter typing rules for CDO custom
+ * connection type checks, but uses the same rules for dragging and safety.
+ * This checker still expects nullable arrays of string for connection
+ * type checks, and still looks for intersections in the arrays. Unlike the
+ * default checker, check arrays that contain a custom connection type are
+ * only compatible with other check arrays with the same connection type.
+ * @implements {Blockly.IConnectionChecker}
+ */
+
+export default class CdoConnectionChecker extends GoogleBlockly.ConnectionChecker {
+  /**
+   * Constructor for the connection checker.
+   */
+  constructor() {
+    super();
+  }
+
+  /**
+   * Check whether this connection is compatible with another connection with
+   * respect to the value type system.  E.g. square_root("Hello") is not
+   * compatible. Includes custom logic for CDO custom connection type checks.
+   *
+   * @param a Connection to compare.
+   * @param b Connection to compare against.
+   * @returns True if the connections share a type.
+   */
```

Comment:
Maybe include in comment that `getCheck` returns an array of types.

Reply:
Done. They can also return `null` which is a big confusing.